webui/react/src/components/TaskBar.tsx

```diff
@@ -34,9 +34,28 @@ export const TaskBar: React.FC<Props> = ({
 }: Props) => {
   const { canModifyWorkspaceNSC } = usePermissions();
   const confirm = useConfirm();
-  const task = useMemo(() => {
-    return { id, name, resourcePool, type } as CommandTask;
-  }, [id, name, resourcePool, type]);
+  const [task, setTask] = useState<CommandTask>();
+
+  const getTaskById = useCallback(async (taskType: CommandType, commandId: string) => {
+    switch (taskType) {
+      case 'command':
+        setTask(await getCommand({ commandId }));
+        break;
+      case 'jupyter-lab':
+        setTask(await getJupyterLab({ commandId }));
+        break;
+      case 'shell':
+        setTask(await getShell({ commandId }));
+        break;
+      case 'tensor-board':
+        setTask(await getTensorBoard({ commandId }));
+        break;
+    }
+  }, []);
+
+  useEffect(() => {
+    getTaskById(type, id);
+  }, [type, id, getTaskById]);
```

Comment:
consider using `useAsync` here -- it wraps the value in a `Loadable` (which we should do here because now it's possible for the task to not load) and handles race conditions:
```suggestion
  const task = useAsync(() => {
    const call = {
      command: getCommand,
      'jupyter-lab': getJupyterLab
      shell: getShell,
      'tensor-board': getTensorBoard
    }[type]
    return call({ commandId: id });
  }, [type, id])
```